aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -99,181 +106,167 @@ The preceding command adds the [aspnet-codegenerator scaffolding tool](xref:fund
-  ```
+A database context class is needed to coordinate EF Core functionality (Create, Read, Update, Delete) for the `Movie` model. The database context is derived from [Microsoft.EntityFrameworkCore.DbContext](xref:Microsoft.EntityFrameworkCore.DbContext) and specifies the entities to include in the data model.
 
-* Run the following command:
+Scaffolding created the following database context class: Data/MvcMovieContext.cs:
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/zDocOnly/MvcMovieContext.cs?name=snippet)]
 
-  ```dotnetcli
-  dotnet aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries
-  ```
+The preceding code creates a [DbSet\<Movie>](/dotnet/api/microsoft.entityframeworkcore.dbset-1) property for the entity set. In Entity Framework terminology, an entity set typically corresponds to a database table. An entity corresponds to a row in the table.
 
-  [!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+<a name="reg"></a>
 
-### [Visual Studio for Mac](#tab/visual-studio-mac)
+### Examine the generated database context registration
 
-* Open a command window in the project directory (The directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files).
+ASP.NET Core is built with [dependency injection (DI)](xref:fundamentals/dependency-injection). Services, such as the EF Core DB context, must be registered with DI in `Startup`. Components that require these services are provided these services via constructor parameters. For example, Razor Pages frequently requires services from the DI container. The constructor code that gets a DB context instance is shown later in the tutorial. In this section, the database context is registered with the DI container.
 
-* Export the scaffold tool path:
+Scaffolding added the following `using` statements at the top of *Startup.cs*:
 
-  ```console
-  export PATH=$HOME/.dotnet/tools:$PATH
-  ```
+```csharp
+using MvcMovie.Data;
+using Microsoft.EntityFrameworkCore;
+```
 
-* Run the following command:
+Scaffolding generated the following highlighted code in `Startup.ConfigureServices`:
 
-  ```dotnetcli
-  dotnet aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries
-  ```
+[!code-csharp[](~/tutorials/first-mvc-app/start-mvc/sample/MvcMovie3/Startup.cs?name=snippet_UseSqlite&highlight=5-6)]
 
-  [!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+The name of the connection string is passed to the context by calling a method on a [DbContextOptions](/dotnet/api/microsoft.entityframeworkcore.dbcontextoptions) object. For local development, the [ASP.NET Core configuration system](xref:fundamentals/configuration/index) reads the `ConnectionString` key from the *appsettings.json* file.
```

Comment:
The `[config system](xref:xyz0` reads the `"MvcMovieContext"` DB connection string. 

> For local development, the [ASP.NET Core configuration system](xref:fundamentals/configuration/index) reads the `ConnectionString` key from the *appsettings.json* file.

Move that to the next sections.